tests/markov/msm/test_msm_basic_properties.py

```diff
@@ -499,3 +508,38 @@ def test_two_state_kinetics(self, setting):
         ksum = 1.0 / t12 + 1.0 / t21
         k2 = 1.0 / t2
         assert_almost_equal(k2, ksum, decimal=3)
+
+
+@pytest.mark.parametrize("k", [0, 1, 2, 4, 6, 10])  #
+@pytest.mark.parametrize("real_eigenvalues", [False, True])  # 0, 1, 2, 5
+def test_msm_power(sparse_mode, k, real_eigenvalues):
+    dim = 50
+    if real_eigenvalues:
+        q = [.5] * dim
+        p = [.5] * dim
+        q[0] = 0
+        p[-1] = 0
+        bdc = BirthDeathChain(q, p, sparse=sparse_mode)
+        transition_matrix = bdc.transition_matrix
+    else:
+        transition_matrix = np.random.uniform(size=(dim, dim))
+        transition_matrix /= np.sum(transition_matrix, axis=1)[..., None]
+        if sparse_mode:
+            transition_matrix = sparse.csr_matrix(transition_matrix)
+    msm = MarkovStateModel(transition_matrix, lagtime=2)
+    assert_(np.all(np.isreal(msm.eigenvalues())) == real_eigenvalues)
+    assert_equal(sparse_mode, msm.sparse)
+    msm_pow = msm ** k
+    assert_equal(msm_pow.lagtime, msm.lagtime * k)
+
+    vec = np.random.uniform(size=(dim,))
+    vec /= vec.sum()
+
+    v = msm.propagate(vec, k)
+    assert_array_almost_equal(v, msm_pow.propagate(vec, 1))
+    if k % msm.lagtime == 0 and real_eigenvalues:
+        assert_equal(v.dtype, msm.transition_matrix.dtype)
+    vec_prop = vec
+    for _ in range(k):
+        vec_prop = msm.transition_matrix.T.dot(vec_prop)
+    assert_array_almost_equal(msm.propagate(vec, k), vec_prop)
```

Comment:
What exactly does this test?

Reply:
manually propagates `vec` by applying `k` times the transition matrix to it vs. using `msm.propagate`